src/Relation/Nullary/Finite/Setoid.agda

```diff
@@ -0,0 +1,148 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Notions of finiteness for setoids
+------------------------------------------------------------------------
+
+{-# OPTIONS --safe --cubical-compatible #-}
+
+module Relation.Nullary.Finite.Setoid where
+
+open import Data.Fin.Base using (Fin)
+open import Data.Nat.Base using (ℕ)
+open import Data.Product.Base using (_×_; _,_; swap; Σ-syntax; proj₁; ∃)
+open import Data.Unit.Base using (⊤)
+open import Function.Base using (id; _∘_)
+open import Function.Bundles using (Func; Injection; Surjection; Inverse)
+open import Level using (Level; _⊔_; 0ℓ) renaming (suc to lsuc)
+open import Relation.Binary.Core using (Rel)
+open import Relation.Binary.Bundles using (Setoid)
+open import Relation.Binary.Structures using (IsEquivalence)
+import Relation.Binary.Construct.On as On
+import Relation.Binary.PropositionalEquality.Properties as ≡
+
+private
+  variable
+    c ℓ c′ ℓ′ : Level
+
+Ω : ∀ p → Setoid (lsuc p) p
+Ω p = record
+  { Carrier = Set p
+  ; _≈_ = λ P Q → (P → Q) × (Q → P)
```

Comment:
This function is [_⇔_](https://agda.github.io/agda-stdlib/Function.Bundles.html#8934). And I just notice a bug: this very same definition is repeated as [_⇔_](https://agda.github.io/agda-stdlib/Function.Equivalence.html#1213) (same name, entirely different file).

In the latter, we also find [⇔-setoid](https://agda.github.io/agda-stdlib/Function.Equivalence.html#2586) which is Omega.

Reply:
I think `Function.Equivalence` is marked for deprecation in v2.0.0; hence, the repetition?

Reply:
Indeed, I missed that part - well spotted!

Reply:
Yeah, `Function.Equivalence`, `Function.Injection`, &c have been pre-deprecated for a long time (since the inception of `Function.Bundles`, probably). Both definitions of `_⇔_` are unfortunate in that the congruence fields become entirely redundant, which is a (post hoc) justification for this much simpler `(P → Q) × (Q → P)` type. As for `⇔-setoid`, maybe I'm biased by the particular use of it here, but it seems like a terrible name. It's not just some random setoid made out of the notion of equivalence; it's the setoid of propositions.

Reply:
I agree, but I don't want this to explode any more than it already has!